akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -303,13 +369,15 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
       byteBuffer.putInt(ClassManifestTagOffset, header._manifestIdx | TagTypeMask)
     else
       writeLiteral(ClassManifestTagOffset, header._manifest)
+
   }
 
   def parseHeader(h: HeaderBuilder): Unit = {
     val header = h.asInstanceOf[HeaderBuilderImpl]
 
     // Read fixed length parts
-    header setVersion byteBuffer.getInt
+    header setVersion byteBuffer.get()
+    header setFlags byteBuffer.get()
     header setUid byteBuffer.getLong
```

Comment:
should read from offset 4: https://github.com/akka/akka/pull/21378/files#diff-82dac4787416496563ec7e2e6c3fe8ecR76